migration/migration.go

```diff
@@ -843,6 +849,42 @@ func setClientSecretIfEmpty(ctx context.Context, mongodb *MongoDB, zenInstance *
 	return
 }
 
+func getHighestRoleForUserFromTeam(userID string, team *v1schema.Team) (role v1schema.Role) {
+	teamUser := team.Users.GetUser(userID)
+	if teamUser == nil {
+		return v1schema.Authenticated
+	}
+	return teamUser.Roles.GetHighestRole()
+}
+
+func setUserV3Role(ctx context.Context, mongodb *MongoDB, user *v1schema.User) (set bool, err error) {
+	if user.Role != "" {
+		return
+	}
+	dbName := "platform-db"
+	collectionName := "Team"
+	filter := bson.D{}
+	cursor, err := mongodb.Client.
+		Database(dbName).
+		Collection(collectionName).
+		Find(ctx, filter)
+
+	var highest v1schema.Role
+	for cursor.Next(ctx) {
+		team := &v1schema.Team{}
+		if err = cursor.Decode(&team); err != nil {
+			return
+		}
+
+		if current := getHighestRoleForUserFromTeam(user.UserID, team); current > highest {
```

Comment:
we need to assess group membership to determine role as well: https://github.ibm.com/IBMPrivateCloud/platform-identity-provider/blob/4.4.0/server/boot/z_cp3migration.js#L453

Reply:
looks like the above is not used. A user may belong to multiple teams.